application/app/Http/Controllers/InstitutionUserController.php

```diff
@@ -147,6 +151,36 @@ function (Builder $query) use ($sortOrder, $sortField) {
         );
     }
 
+    /** @throws AuthorizationException|Throwable */
+    public function activate(ActivateInstitutionUserRequest $request): InstitutionUserResource
+    {
+        return DB::transaction(function () use ($request) {
+            /** @var $institutionUser InstitutionUser */
+            $institutionUser = $this->getBaseQuery()
+                ->withoutGlobalScope(ExcludeDeactivatedInstitutionUsersScope::class)
+                ->findOrFail($request->validated('institution_user_id'));
+
+            $this->authorize('activate', $institutionUser);
+
+            $institutionUser->deactivation_date = null;
+            $institutionUser->roles()->sync(
+                Role::query()
+                    ->withoutGlobalScope(RolePolicy::scope())
+                    ->findMany($request->validated('roles'))
+            );
+            $institutionUser->saveOrFail();
```

Comment:
It was actually wrapped in `DB::transaction()`, so any updates were batched together, but I implemented your suggestion anyway. :)

Reply:
Updates inside transaction are still against the database, transaction just commits those updates to make them permanent or rollbacks them. Either way subsequent actions inside the transaction are still influenced by the previous ones, e.g
```
$bIds = [...];
$a = new A;
$a->b()->sync($bIds);
```
This fails since sync tries to create a pivot table row which references B id and A id and since A haven't been saved before then fetching ID reference of A object inside sync method internally fails